### PR TITLE
Feat/2646 total errors aggregates

### LINKED ISSUE #2646: Populate `DataFileSummary.case_aggregates` differently for each section
**Description:**

According to the [UX HTML file](https://reitermb.github.io/HTML-Table-example/) different datafile submission sections have different aggregate data displayed in the table. 

Sections 01 and 02
image

Sections 03 and 04
image

To facilitate this, we need the ability to run a different aggregation function for different submission sections. #2612 [introduces `dfs.case_aggregates = case_aggregates_by_month(data_file)`](https://github.com/raft-tech/TANF-app/blob/d3bceb7da6f43c3c8b30e3be13b9c0412399795e/tdrs-backend/tdpservice/scheduling/parser_task.py#L27) which will need to be repurposed and replicated for the different sections.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] A `DataFileSummary` created for a submission has `case_aggregates` populated with the correct data for the section
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Modify `scheduling/parser_task.py` similar to the example implementation below, running a different aggregation function for different sections and populating `dfs.case_aggregates` appropriately
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
Example implementation (in `parser_task.py`)

```python
def case_aggregates_by_section(data_file):
    if data_file.section.contains('Active Case Data'):
        return case_aggregates_by_month(data_file)
    elif data_file.section.contains('Closed Case Data'):
        return # some other function call
    # ... repeat for all sections

def parse(data_file_id)
    # ...
    dfs = DataFileSummary.objects.create(datafile=data_file, status=DataFileSummary.Status.PENDING)
    errors = parse_datafile(data_file)
    dfs.status = dfs.get_status(errors=errors)
    dfs.case_aggregates = case_aggregates_by_section(data_file)
    dfs.save()
```

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- [_HTML copy_](https://reitermb.github.io/HTML-Table-example/)
- _Doc 2_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- 1, 2, 3, 4 all different or 1/2 and 3/4?
- Existing tickets for sections 2/3/4?
- Frontend impl?

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes
Pull request closes #2646 

* Implements a new `total_errors_by_month` aggregation function, implemented for section 3+4 submissions
* Updates the frontend `SubmissionHistory` component structure to be more readable and limit conditional rendering. some small code duplications and inefficiencies can be ironed out if necessary

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Submit data files for all sections to test the frontend restructuring
1. Submit data files for sections 3+4 and validate the "Total Errors" column in the submission history

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] A DataFileSummary created for a submission has `case_aggregates` populated with the correct data for the section
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/parsers/test/test_parse.py
Comment (elipe17) [inline tdrs-backend/tdpservice/parsers/test/test_parse.py:941]: Is there a way we can fudge one of these files, or make a new one, to capture the case where we have errors across our split out records?

Comment (jtimpe) [reply tdrs-backend/tdpservice/parsers/test/test_parse.py:941]: yep! good callout

File: tdrs-backend/tdpservice/parsers/test/test_parse.py
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/test/test_parse.py]: nit: do you need to save the dfs? Normally the transaction is off so .save() should not matter

File: tdrs-backend/tdpservice/scheduling/parser_task.py
Comment (raftmsohani) [inline tdrs-backend/tdpservice/scheduling/parser_task.py:28]: not related to this PR but we should have had "Case Data" as a CONSTANT, and I would put it in .DataFile.py

File: tdrs-backend/tdpservice/parsers/util.py
Comment (jtimpe) [inline tdrs-backend/tdpservice/parsers/util.py]: move to `aggregators.py` once 2711 merges

File: tdrs-backend/tdpservice/parsers/util.py
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/parsers/util.py]: Is this your auto-formatter? Looks odd

File: tdrs-backend/tdpservice/parsers/util.py
Comment (andrew-jameson) [reply tdrs-backend/tdpservice/parsers/util.py]: 2711 merged earlier this morning, you can now pull in latest from develop

File: tdrs-backend/tdpservice/parsers/util.py
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/parsers/util.py]: What edge case is this catching?

File: tdrs-backend/tdpservice/parsers/test/test_parse.py
Comment (jtimpe) [reply tdrs-backend/tdpservice/parsers/test/test_parse.py]: they were just holdovers from the old tests. i removed them and the tests still pass, thanks!

**Review by elipe17 — APPROVED** [review]
Looks good!

File: tdrs-backend/tdpservice/parsers/aggregates.py
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/aggregates.py]: we should be able to query all ParserErrors before running the for loop, and have the counts in a table.
 
Another note: as we are parsing more and more files, this query would take forever to run, the problem is df is indexed easily but the rpt_month_year will be slow.

**Review by raftmsohani — CHANGES_REQUESTED** [review]
more efficient query!

**Comment by ADPennington** [comment]
@jtimpe did this fall off my radar?

**Comment by jtimpe** [comment]
> @jtimpe did this fall off my radar?

nope! i am still addressing PR feedback and getting approvals - once i've done so i'll follow back up with you

File: tdrs-backend/tdpservice/parsers/aggregates.py
Comment (jtimpe) [reply tdrs-backend/tdpservice/parsers/aggregates.py]: i pulled the main `ParserError.objects.all()` out of the loop - are you suggesting grouping by rpt_month_year or something to that effect as well?

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Review by andrew-jameson — APPROVED** [review]
Good to go!

**Comment by ADPennington** [comment]
@jtimpe there is a merge conflict here. can you ping me when its ready again?

**Review by ADPennington — APPROVED** [review]
lgtm @jtimpe 🚀 

- Section 3 and 4 total errors for tanf, tribal, and ssp get populated with 0 or non-zero values when status != `Rejected`
- there are a couple of edge cases that I presume will be addressed in later tickets (let me know if this is not the case):
  - `Accepted with errors` status and only error detected is in trailer record: Total Errors == 0 :warning: (maybe [Cat1 Cleanup](https://github.com/raft-tech/TANF-app/pull/2838) will resolve this )
  - any `Partially Accepted with Errors` status: Total Errors == 0 – this status only comes up when at least one data record (not header/trailer) is not correct length to parse. stratum and aggregate data files only have one data record. so this status does not make sense. :warning: will be addressed in [#2754](https://github.com/raft-tech/TANF-app/issues/2754)

test notes [here](https://hackmd.io/P4sQkTRORLGjiodfgRwMtA?view#Feat2646-total-errors-aggregates)

**Pull request merged**
